File: guides/micronaut-graphql-todo/kotlin/src/main/kotlin/example/micronaut/ToDo.kt
package example.micronaut

import io.micronaut.data.annotation.GeneratedValue
import io.micronaut.data.annotation.GeneratedValue.Type.AUTO
import io.micronaut.data.annotation.Id
import io.micronaut.data.annotation.MappedEntity

@MappedEntity // <1>
class ToDo(var title: String, val authorId: Long) {

    @Id // <2>
    @GeneratedValue(AUTO)
    var id: Long? = null

    var isCompleted = false
}

File: guides/micronaut-graphql-todo/python/src/example/micronaut/to_do.py
from dataclasses import dataclass
from typing import Annotated

from jakarta.validation.constraints import NotNull
from micronaut.core.annotation import Introspected
from micronaut.data.annotation import GeneratedValue, Id, MappedEntity


@dataclass
@Introspected
@MappedEntity  # <1>
class ToDo:
    title: Annotated[str, NotNull]
    authorId: int
    completed: bool = False
    id: Annotated[int | None, Id, GeneratedValue] = None  # <2>
